source/solution_deploy/lib/remediation_runbook-stack.ts

```diff
@@ -45,1299 +45,1480 @@ export interface MemberRoleStackProps {
-                    id: 'W12',
-                    reason: 'Resource * is required for to allow remediation for *any* resource.'
-                }]
+            let childToMod = inlinePolicy.node.findChild('Resource') as CfnPolicy;
+            childToMod.cfnOptions.metadata = {
+                cfn_nag: {
+                    rules_to_suppress: [{
+                        id: 'W12',
+                        reason: 'Resource * is required for to allow remediation.'
+                    }, {
+                        id: 'W28',
+                        reason: 'Static names chosen intentionally to provide integration in cross-account permissions.'
+                    }]
+                }
             }
         }
-    }
 
-    //-----------------------
-    // EnableCloudTrailToCloudWatchLogging
-    //
-    {
-        const remediationName = 'EnableCloudTrailToCloudWatchLogging'
-        const inlinePolicy = new Policy(props.roleStack, `SHARR-Remediation-Policy-${remediationName}`);
-
-        // Role for CT->CW logging
-        const ctcw_remediation_policy_statement_1 = new PolicyStatement()
-        ctcw_remediation_policy_statement_1.addActions("logs:CreateLogStream")
-        ctcw_remediation_policy_statement_1.effect = Effect.ALLOW
-        ctcw_remediation_policy_statement_1.addResources(
-            "arn:" + this.partition + ":logs:*:*:log-group:*"
-        )
-
-        const ctcw_remediation_policy_statement_2 = new PolicyStatement()
-        ctcw_remediation_policy_statement_2.addActions("logs:PutLogEvents")
-        ctcw_remediation_policy_statement_2.effect = Effect.ALLOW
-        ctcw_remediation_policy_statement_2.addResources(
-            "arn:" + this.partition + ":logs:*:*:log-group:*:log-stream:*"
-        )
-
-        const ctcw_remediation_policy_doc = new PolicyDocument()
-        ctcw_remediation_policy_doc.addStatements(ctcw_remediation_policy_statement_1)
-        ctcw_remediation_policy_doc.addStatements(ctcw_remediation_policy_statement_2)
-
-        const ctcw_remediation_role = new Role(props.roleStack, 'ctcwremediationrole', {
-            assumedBy: new ServicePrincipal(`cloudtrail.${this.urlSuffix}`),
-            inlinePolicies: {
-                'default_lambdaPolicy': ctcw_remediation_policy_doc
-            },
-            roleName: `${RESOURCE_PREFIX}-CloudTrailToCloudWatchLogs`
-        });
-        ctcw_remediation_role.applyRemovalPolicy(cdk.RemovalPolicy.RETAIN)
+        //-----------------------
+        // CreateIAMRole
+        //
         {
-            let childToMod = ctcw_remediation_role.node.findChild('Resource') as CfnRole;
+            const remediationName = 'CreateIAMRole'
+            const inlinePolicy = new Policy(props.roleStack, `SHARR-Remediation-Policy-${remediationName}`);
+            const iamPerms = new PolicyStatement();
+            iamPerms.addActions(
+                "iam:GetRole",
+                "iam:CreateRole",
+                "iam:AttachRolePolicy",
```

Comment:
A role that can attach policies should exclude itself so it can't escalate its own permissions.